planning/behavior_path_planner/src/utils/path_utils.cpp

```diff
@@ -69,10 +69,17 @@ PathWithLaneId resamplePathWithSpline(
   }
 
   constexpr double epsilon = 0.2;
-  const auto has_almost_same_value = [&](const auto & vec, const auto x) {
-    if (vec.empty()) return false;
-    const auto has_close = [&](const auto v) { return std::abs(v - x) < epsilon; };
-    return std::find_if(vec.begin(), vec.end(), has_close) != vec.end();
+  const auto is_close = [&](double v, double x) { return std::abs(v - x) < epsilon; };
+  const auto find_almost_same_value = [&](const std::vector<double> & vec, double x) {
+    if (vec.empty()) return boost::optional<size_t>();
+
+    auto it = std::find_if(vec.begin(), vec.end(), [&](double v) { return is_close(v, x); });
```

Comment:
```suggestion
    const auto it = std::find_if(vec.begin(), vec.end(), [&](double v) { return is_close(v, x); });
```

Reply:
thanks! fix in [const](https://github.com/autowarefoundation/autoware.universe/pull/3873/commits/076ccaef8969d7b993834a20a182fce2163fcab8)
but change algorithm for multiple close point in [cosider multiple close point](https://github.com/autowarefoundation/autoware.universe/pull/3873/commits/6e3b35fc4a697b48c340ae524b6683b5d030e140)